case-lib/lib.sh

```diff
@@ -810,9 +810,31 @@ is_firmware_file_zephyr()
     test "$znum" -gt 10
 }
 
+is_ipc4_debugfs()
+{
+    local ipc_type
+
+    ipc_file=/sys/kernel/debug/sof/ipc_type
+    ipc_type=$(cat $ipc_file) || {
+        return 1
+    }
```

Comment:
the overloading is already present in the initial code for is_ipc4().

And while I am at it it's just torture to see code a branch " if is_ipc4" taken when ipc4 is NOT used. 
Such mind-twisting is just not right. someone will screw-up when the value of 'true' is inverted.

Reply:
> the overloading is already present in the initial code for is_ipc4().

It's not because the higher level does something questionable that the lower level, internal implementation must align with it. Of course it shouldn't be different without any good reason either but there's no "strict" requirement there.